### ISSUE TITLE
Optional tests (for flaky or "nice to have but not required" tests)

### ISSUE THREAD

**Issue by fregante** [vitest-dev/vitest#4920]
### Clear and concise description of the problem

As the title says, it might be nice to run and output the results of a test, but ignore them if they fail (e.g. don't set the `exitCode` to 1)

In my case, the tests require a network connection towards a third-party server. Mocking defeats the purpose of these tests because they specifically test the support of the third party.

### Suggested solution

Examples:

- `describe.optional`
- `test.optional`
- `/* @vitest-optional */` in header

### Alternative

```sh
npm run vitest -- test/optional || echo 'Optional tests failed'
npm run vitest -- --exclude test/optional
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by fregante** [comment]
Unfortunately it doesn't look like it varies over (a short period of) time. 


I may agree that it's not a great pattern, but flaky tests are very common in my experience. What people end up doing is just disabling/skipping the tests altogether, forever. I'm not sure that that's a better outcome. 

With a running-but-optional you at least get some results and annotations (#4921)

**Comment by sheremet-va** [comment]
I don't think this is something a test runner should endorse. I would expect users to handle such situations themselves. For example if an API fail is not considered a failure, then it might be a good idea to accept that as a valid result:

```ts
function safeFetch(url: string) {
  try {
    const result = await fetch(url)
    return { status: 'ok', result: await result.json() }
  } catch (err) {
    return { status: 'error', result: err }
  }
}

test('api call', async () => {
  expect.assertions(1)
  const { status, result } = await safeFetch('/api')
  if (status === 'ok') {
    expect(result).toEqual({ index: 1 })
  } else {
    expect(result.message).toMatch(/cannot establish connection/)
  }
})
```

**Comment by fregante** [comment]
> handle such situations themselves

This isn't possible. You can't both output the results of the failed expectations *and* have the test pass. If I handle it myself, the test will silently pass 100% of the time, regardless of status. That defeats the purpose of using a test runner because I might as well write `expect(true).toBe(true)`

> I don't think this is something a test runner should endorse.

Kinda agree kinda not. It endorses tests that fail 100% of the time already.
